app/controllers/authenticate_controller.rb

```diff
@@ -1,30 +1,78 @@
 class AuthenticateController < ApplicationController
   include TokenGenerator
 
-  # Request path indicates the username for +authenticate+.
-  before_filter :credentials_lookup
-
   def authenticate
-    api_key = request.body.read
-    if @credentials.valid_api_key? api_key
-      render json: authentication_token
+
+    authenticator = params[:authenticator]
+    service_id    = params[:service_id]
+    account       = params[:account]
+    username      = params[:id]
+    password      = request.body.read
+    role_id       = Role.roleid_from_username(account, username)
+
+    validate_role_exists!(role_id)
+    
+    case authenticator
+    when 'authn-ldap'
+      validate_security!(authenticator, account, service_id, username)
+      raise Unauthorized unless ldap_authenticator.valid?(username, password)
+      role = Role[role_id]
+    # default conjur: authn
     else
```

Comment:
@jonahx it should check specifically for `when 'authn'` and return 401 if they do not return an authenticator that is valid